code/modules/library/book.dm

```diff
@@ -0,0 +1,370 @@
+
+/**
+  * Show Content Proc
+  *
+  * Builds the browserUI html to show to the player then open up the User interface. It first build the header navigation
+  * buttons and then builds the rest of the UI based on what page the player is turned to.
+  */
+/obj/item/book/proc/show_content(mob/user)
+	var/dat = " "
+	//First, we're going to choose/generate our header buttons for switching pages and store it in var/dat
+	if(length(pages)) //No need to have page switching buttons if there's no pages
+		if(current_page == 0)
+			dat+= "<DIV STYLE='float:left; text-align:left; width:49.9%'></DIV>"
+			dat+= "<DIV STYLE='float:left; text-align:right; width:49.9%'><A href='?src=[UID()];next_page=1'>Next Page</A></DIV><BR><HR>"
+		else if(current_page < length(pages))
+			dat+= "<DIV STYLE='float:left; text-align:left; width:49.9%'><A href='?src=[UID()];prev_page=1'>Previous Page</A></DIV>"
+			dat+= "<DIV STYLE='float:left; text-align:right; width:49.9%'><A href='?src=[UID()];next_page=1'>Next Page</A></DIV><BR><HR>"
+		else if(current_page == length(pages))
+			dat+= "<DIV STYLE='float:left; text-align:left; width:49.9%'><A href='?src=[UID()];prev_page=1'>Previous Page</A></DIV>"
+			dat+= "<DIV STYLE='float;left; text-align:right; width:49.9%'></DIV>"
+
+	//Now we're going to display the header buttons + the current page selected, if it's page 0, we display the cover_page instead
+	if(!current_page)
+		var/cover_page = {"<center><h1>[title]</h1><br></h2>Written by: [author]</h2></center><br><hr><b>Summary:</b> [summary]"}
+		user << browse("<body bgcolor='[book_bgcolor]'>[dat]<BR>" + "[cover_page]", "window=book[UID()];size=400x400")
+		return
+	else
+		user << browse("<body bgcolor='[book_bgcolor]'>[dat]<BR>" + "[pages[current_page]]", "window=book[UID()]")
+
+/obj/item/book/Topic(href, href_list)
+	..()
+	if(loc != usr && !Adjacent(usr))
+		return
+	if(href_list["next_page"])
+		if(current_page > length(pages)) //should never be false, but just in-case
+			current_page = length(pages)
+			return
+		current_page++
+		playsound(loc, "pageturn", 50, 1)
+		attack_self(usr) //scuffed but this is how you update the UI
+		updateUsrDialog()
+	if(href_list["prev_page"])
+		if(current_page < 0) //should never be false, but just in-case
+			current_page = 0
+			return
+		current_page--
+		playsound(loc, "pageturn", 50, 1)
+		attack_self(usr) //scuffed but this is how you update the UI
+		updateUsrDialog()
+
+/**
+  * Edit Book Proc
+  *
+  * This is where a lot of the magic happens, upon interacting with the book with a pen, this proc will open up options
+  * for the player to edit the book. Most importantly, this is where we account for player stupidity and maliciousness
+  * any input must strip/reject bad text and HTML from user input, additionally we account for players trying to screw
+  * with the Database. This will also limit the max characters players can upload at one time to prevent spamming.
+  */
+/obj/item/book/proc/edit_book(mob/user)
```

Comment:
`user` is passed here but is not used in most cases down below. Be sure to actually use `user` instead of `usr`. Also for the `input()` proc

Reply:
should be fixed now